### ISSUE TITLE
support to store virtual machines in filesystem

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#58]
Originally reported of Google Code with ID 1.

```
WISHLIST: 
Add support for using filesystem as backend for virtual machines block
devices. 

If you have a shared filesystem between the physical servers and store the
virtual machines in this filesystem, you can do live migration. 

Ganeti doesn't need to know what filesystem is used to store the images,
just need to use the posix filesystem interface. So you can use any posix
compliant filesystem to store the image. 

One example would be:  
You have a glusterFS filesystem available for all your physical servers and
this filesystem is mounted in /xen/ 

Ganeti will create vm images in filesystem directory structure. like:
/xen/vm01/disk.img and /xen/vm01/swap.img 



```
Originally added on 2007-08-30 16:09:31 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Hi,

We have considered implementing file-based backend, however it requires many changes
across the ganeti codebase, so it's not possible to do it in the 1.2 timeframe.

We will most likely support some version of file storage for the next version.
```
Originally added on 2007-08-31 11:27:08 +0000 UTC.

Added Labels: Type-Enhancement
Removed Labels: Type-Defect
Changed State: Accepted
Added to Milestone: Release1.3

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2008-01-17 10:27:55 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
File-backend instances are on the way. For more information see the corresponding
wiki-page.

Cheers,
Manuel
```
Originally added on 2008-01-22 16:04:05 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Note: this has been implemented, and will be released in the 2.0 version.
```
Originally added on 2008-07-03 18:22:30 +0000 UTC.

Changed State: Started
Added to Milestone: Release2.0
Removed from Milestone: Release1.3

**Comment by Ganeti-Issues-Migrator** [comment]
```
This was released with 2.0.
```
Originally added on 2009-08-05 08:50:25 +0000 UTC.

Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2012-10-22 08:57:06 +0000 UTC.

Changed State: Released
